crates/wash-lib/src/build/provider.rs

```diff
@@ -75,62 +147,73 @@ pub(crate) async fn build_provider(
     if let Some(override_target) = &provider_config.rust_target {
         provider_path_buf.push(override_target);
     }
+
     provider_path_buf.push("release");
     provider_path_buf.push(&bin_name);
 
-    trace!("Retrieving provider binary from {:?}", provider_path_buf);
+    Ok((provider_path_buf, bin_name))
+}
 
-    let provider_bytes = tokio::fs::read(&provider_path_buf).await?;
+/// Build a Go provider for the current machine's architecture
+///
+/// Returns a tuple of the path to the built provider binary and the binary name
+async fn build_go_provider(
+    provider_config: &ProviderConfig,
+    go_config: &TinyGoConfig,
+    common_config: &CommonConfig,
+) -> Result<(PathBuf, String)> {
+    let mut generate_command = match go_config.go_path.as_ref() {
+        Some(path) => process::Command::new(path),
+        None => process::Command::new("go"),
+    };
 
-    let mut par = create_provider_archive(
-        ParCreateArgs {
-            vendor: provider_config.vendor.to_string(),
-            revision: Some(common_config.revision),
-            version: Some(common_config.version.to_string()),
-            schema: None,
-            name: common_config.name.to_string(),
-            arch: detect_arch(),
-        },
-        &provider_bytes,
-    )
-    .context("failed to create initial provider archive with built provider")?;
+    // Change directory into the project directory
+    std::env::set_current_dir(&common_config.path)?;
+    trace!("Building provider in {:?}", common_config.path);
 
-    // If no signing config supplied, just return the path to the provider
-    let Some(sign_config) = signing_config else {
-        warn!("No signing configuration supplied, could only build provider");
-        return Ok(provider_path_buf);
-    };
+    // Generate interfaces, if not disabled
+    if !go_config.disable_go_generate {
```

Comment:
I think, even though it's weirder in the code, that disabling it explicitly is nicer since it lets people not specify the field and have it default to generate

Reply:
I was thinking we could enable it but default to true -- I'm not too particular about this here in the interest of :rocket: